src/minimum_tracer/minimum_tracer.cpp

```diff
@@ -0,0 +1,3532 @@
+  Hessian_2 = [=](auto const &arg)
+  { return HessianNumerical(arg, V_2, eps, dim); };
+
+  // Initial guess for middle point
+  point_m = point_1;
+
+  // Calculate smallest EV
+  ev_1 = SmallestEigenvalue(point_1, Hessian_1);
+  ev_2 = SmallestEigenvalue(point_2, Hessian_2);
+
+  if (ev_1 * ev_2 > 0)
+  {
+    // Safety check. We should never be here.
+    std::stringstream ss;
+    ss << "They have the same sign, no zero to be found!\t" << ev_1 << "\t"
+       << ev_2 << "\n";
+    Logger::Write(LoggingLevel::MinTracerDetailed, ss.str());
+    point_1.push_back(-1);
+    return point_1;
+  }
+
+  // This is to provent instabilities and make sure we do not leave the
+  // phase
+  double AllowedMaxDistance = fmax(1, L2NormVector(point_1 - point_2) / dim);
+  std::vector<double> CandidatePoint = point_1;
+  double CandidateTemperature        = T_1;
+
+  while (abs(T_1 / T_2 - 1) > 1e-8)
+  {
+    T_m = (T_1 + T_2) / 2.;
+    // Define potential in the middle
+    V_m = [&](std::vector<double> vev)
+    {
+      // Potential wrapper
+      std::vector<double> res = this->modelPointer->MinimizeOrderVEV(vev);
+      return this->modelPointer->VEff(res, T_m) / (1 + T_m * T_m);
+    };
+    dV_m = [=](auto const &arg) { return NablaNumerical(arg, V_m, eps, dim); };
+    Hessian_m = [=](auto const &arg)
+    { return HessianNumerical(arg, V_m, eps, dim); };
+    point_m =
+        LocateMinimum(point_m, dV_m, Hessian_m, 1e-3 * dim / (1 + T_m * T_m));
+    ev_m = SmallestEigenvalue(point_m, Hessian_m);
+
+    // Check if there is numerical instabilities so reduce the searching area.
+    if (L2NormVector(point_m - point_2) / dim > AllowedMaxDistance or
+        L2NormVector(point_1 - point_m) / dim > AllowedMaxDistance)
+    {
+      T_2 = T_m;
+      // break;
+    }
+    else if (ev_1 * ev_m > 0)
+    {
+      // This point has the correct eigenvalue and is close.
+      CandidatePoint       = point_m;
+      CandidateTemperature = T_m;
+    }
+
+    if (ev_1 * ev_m > 0)
+    {
+      T_1 = T_m;
+    }
+    else if (ev_2 * ev_m > 0)
+    {
+      T_2 = T_m;
+    }
+    else
+    {
+      // Here ev_m == 0
+      break;
+    }
+  }
+  CandidatePoint.push_back(CandidateTemperature);
+  return CandidatePoint;
+}
+
+/**
+ * @brief TrackPhase with enforced global minimum tracing (= phase is checked if
+ * it is still the global minimum until it is no longer, then the current
+ * temperature is stored in globMinEndT)
+ * @param globMinEndT temperature at which phase is no longer global minimum
+ * @param point start point for tracking
+ * @param currentT start point temperature for phase tracking
+ * @param finalT end point temperature
+ * @param dT initial temperature step size
+ * @param output if true tracking output is printed on the screen
+ * @param unprotected if true we dont check the hessian
+ */
+std::vector<Minimum> MinimumTracer::TrackPhase(double &globMinEndT,
+                                               std::vector<double> point,
```

Comment:
```suggestion
                                               const std::vector<double>& point,
```

Reply:
This cannot be implemented. We use/change ```point``` in the code.

Reply:
Usually I would prefer then to make a local copy in the function and edit this. Then you could spare the copy in case of an early exit but this is fine for now